lib/elixir/lib/task/supervised.ex

```diff
@@ -539,9 +531,55 @@ defmodule Task.Supervised do
 
       {:EXIT, _, _} ->
         stream_monitor_loop(running_tasks, config)
+
+      other ->
+        handle_stop_or_parent_down(other, running_tasks, config)
+        stream_monitor_loop(running_tasks, config)
     end
   end
 
+  defp stream_waiting_for_stop_loop(running_tasks, config) do
+    %{
+      parent_ref: parent_ref,
+      monitor_ref: monitor_ref
+    } = config
+
+    receive do
+      message ->
```

Comment:
Just to double check: this was matching before https://github.com/elixir-lang/elixir/pull/8239/commits/bbef09885a0ea8ba0e12fe82786373afffb68fce on:

* `{:stop, ^monitor_ref}`
* `{:DOWN, ^parent_ref, _, _, reason}`
* `_other` (match all)

However, I think it now only matches on:

* `{:stop, ^monitor_ref}` (the first clause of `handle_stop_or_parent_down`) 
* `{:DOWN, ^parent_ref, _, _, reason}` (the second clause of `handle_stop_or_parent_down`)

The previous match-all is gone. Could that produce undesired `FunctionClauseError`s?

Reply:
It will lead to undefined FunctionClauseError. We need to ignore the message in case it is not the one we are looking for.

Reply:
The problem with this is that before the `stream_monitor_loop` didn't accept random messages as the catch-all clause was `{:EXIT, ...}`. Not sure how to approach this.

Reply:
You can make `handle_stop_or_parent_down` return true (if handled) or false (if unknown) and raise accordingly. `handle_stop... || raise "unexpected message #{inspect other}"`.

Reply:
Yeah that makes sense. I'm now raising in `stream_monitor_loop` but not in `stream_waiting_for_stop_loop`. Should work :)